resolwe_bio/variants/models.py

```diff
@@ -0,0 +1,151 @@
+GENOME_ASSEMBLY_MAX_LENGTH = 20
+CHROMOSOME_MAX_LENGTH = 20
+REFERENCE_MAX_LENGTH = 100
+ALTERNATIVE_MAX_LENGTH = 100
+
+# Metadata
+VARIANT_DATA_SOURCE_MAX_LENGTH = 100
+VARIANT_ANNOTATION_SOURCE_MAX_LENGTH = 100
+ANNOTATION_MAX_LENGTH = 200
+GENOTYPE_MAX_LENGTH = 100
+TYPE_MAX_LENGTH = 100
+CLINICAL_DIAGNOSIS_MAX_LENGTH = 200
+CLINICAL_SIGNIFICANCE_MAX_LENGTH = 100
+DBSNP_ID_MAX_LENGTH = 20
+CLINICAL_VAR_ID_MAX_LENGTH = 20
+ANNOTATION_IMPACT_MAX_LENGTH = 20
+GENE_MAX_LENGTH = 100
+PROTEIN_IMPACT_MAX_LENGTH = 100
+FEATURE_ID_MAX_LENGTH = 200
+FILTER_MAX_LENGTH = 20
+
+
+class Variant(models.Model):
+    """Describe a variant in the database."""
+
+    # Because Django ORM cannot handle composite primary keys, each feature is
+    # still assigned an internal numeric 'id' and the ('source', 'feature_id',
+    # 'species') combination is used to uniquely identify a feature.
+    species = models.CharField(max_length=SPECIES_MAX_LENGTH)
+    genome_assembly = models.CharField(max_length=GENOME_ASSEMBLY_MAX_LENGTH)
+    chromosome = models.CharField(max_length=CHROMOSOME_MAX_LENGTH)
+    position = models.PositiveBigIntegerField()
+    reference = models.CharField(max_length=REFERENCE_MAX_LENGTH)
+    alternative = models.CharField(max_length=ALTERNATIVE_MAX_LENGTH)
+
+    class Meta:
+        """Add constraint for composite key."""
+
+        constraints = [
+            models.UniqueConstraint(
+                fields=[
+                    "species",
+                    "genome_assembly",
+                    "chromosome",
+                    "position",
+                    "reference",
+                    "alternative",
+                ],
+                name="uniq_composite_key_variants",
+            ),
+        ]
+
+
+class VariantAnnotation(models.Model):
+    """Describes an annotation of a variant."""
+
+    variant = models.OneToOneField(
+        Variant, on_delete=models.CASCADE, related_name="annotation"
+    )
+    type = models.CharField(max_length=TYPE_MAX_LENGTH, blank=True, null=True)
+    annotation = models.CharField(max_length=ANNOTATION_MAX_LENGTH)
+    annotation_impact = models.CharField(max_length=ANNOTATION_IMPACT_MAX_LENGTH)
+    gene = models.CharField(max_length=GENE_MAX_LENGTH)
+    protein_impact = models.CharField(max_length=PROTEIN_IMPACT_MAX_LENGTH)
+    feature_id = ArrayField(
+        models.CharField(max_length=FEATURE_ID_MAX_LENGTH), default=list
+    )
+    clinical_diagnosis = models.CharField(
+        max_length=CLINICAL_DIAGNOSIS_MAX_LENGTH, blank=True, null=True
+    )
+    clinical_significance = models.CharField(
+        max_length=CLINICAL_SIGNIFICANCE_MAX_LENGTH, blank=True, null=True
+    )
+    # Optional references to the external databases.
+    dbsnp_id = models.CharField(max_length=DBSNP_ID_MAX_LENGTH, blank=True, null=True)
+    clinical_var_id = models.CharField(
+        max_length=CLINICAL_VAR_ID_MAX_LENGTH, blank=True, null=True
+    )
+    data = models.ForeignKey(
```

Comment:
Is it ok to delete these data when the Data object is deleted, but the sample may still exist?